turbinia/evidence.py

```diff
@@ -369,8 +375,60 @@ class FilteredTextFile(TextFile):
 class ExportedFileArtifact(Evidence):
   """Exported file artifact."""
 
-  def __init__(self, artifact_name):
+  def __init__(self, artifact_name=None):
     """Initializes an exported file artifact."""
     super(ExportedFileArtifact, self).__init__()
     self.artifact_name = artifact_name
     self.copyable = True
+
+
+class DockerContainer(Evidence):
+  """Evidence object for a DockerContainer filesystem.
+
+  Attributes:
+    container_id(str): The ID of the container to mount.
+  """
+
+  # ABSOLUTELY NO LEADING / HERE
+  DEFAULT_DOCKER_DIRECTORY_PATH = 'var/lib/docker'
+
+  def __init__(self, container_id=None, *args, **kwargs):
+    """Initialization for Docker Container."""
+    super(DockerContainer, self).__init__(*args, **kwargs)
+    self.container_id = container_id
+    self._container_fs_path = None
+    self._docker_root_directory = None
+    self._mount_path = None
+    self.context_dependent = True
+
+  def _preprocess(self):
+    if not self.parent_evidence:
+      raise TurbiniaException(
+          'Evidence of type DockerContainer should have a parent_evidence set.')
+    self._mount_path = self.parent_evidence.local_path
+    if issubclass(type(self.parent_evidence), RawDisk):
+      # Mounting the filesystem on the disk, as this is not done by the RawDisk
+      # evidence preprocessor
+      # TODO: consider making the RawDisk preprocessor always mount the
+      # partition
+      self._mount_path = mount_local.PreprocessMountDisk(
+          self.parent_evidence.loopdevice_path,
+          self.parent_evidence.mount_partition)
+
+    self._docker_root_directory = os.path.join(
```

Comment:
So, right now for GoogleCloudDisk the `local_path` refers to the block device and not the mounted dir (and that gets assigned to `_mount_path` above), so I don't think this will work for that evidence type as a parent. :(.  I think as part of moving the mount calls into the RawDisk/GoogleCloudDisk preprocessors (see below comment for more), we should maybe disambiguate these into `device_path` and the already existing `mount_path` (from RawDisk), and for now we can just use `mount_path` of the parent here instead.

Long term, I'm not sure if we should keep `local_path` then or not.  We could keep it for Tasks that know how to process both (e.g. Plaso), or we could just remove it altogether and force the Task writers to explicitly use one or the other.  A quick grep shows 92 references to local_path across 20 files, so this won't be trivial though (so definitely a separate PR).

Reply:
I think "local_path" should be the only thing that's 'visible' to tasks trying to access the data pointed to by the Evidence, and all the other paths should just be hidden & used by the pre & post processors for the management of the Evidence setup/teardown

Reply:
Okay, the "mountening" thing needs to be its own PR :(
See https://github.com/google/turbinia/pull/385